pallets/account-manager/src/manager.rs

```diff
@@ -46,6 +59,89 @@ impl<T: Config> AccountManagerExt<T::AccountId, BalanceOf<T>> for Pallet<T> {
         Ok(())
     }
 
+    /// Reward executor for successful sfx execution - accounted after successful xtx resolution
+    fn commit_charge(executor: T::AccountId, sfx_id: T::Hash) -> DispatchResult {
+        let pending_charge: SfxRequestCharge<T::AccountId, BalanceOf<T>> =
+            if let Some(pending_charge) =
+                SfxPendingChargesPerRound::<T>::get(T::CircuitClock::current_round(), sfx_id)
```

Comment:
`SfxPendingChargesPerRound` converts to `SfxSettlement` only after externally called by Circuit. This must only happen after the Xtx is successfully resolved - if COMMIT then call for `commit_charge`, if REVERT then `refund_charge`. 

The funds transfer to executor is designed to happen asynchronously only at the time executor / staked is ready to `claim` - global clock must implement it (todo). That's a good business value proposition since is in the full control of when would the transfers appear in the context of taxable events etc.

Reply:
Ah so for point one, perhaps that should be some storage in the circuit, then and only once it is ready to be committed to the account manager, you should move it as a ledger item to be paid.

This does a few things:
- avoids inconsistency with data
- decouples the account manager from circuit and keeps the responsibility of only managing stuff that is going to be paid
- avoids polluting the account manager with very specific storage items

Another option, since i see your point about having some "promotable" ledger items, that is, an item that should not be issues until some rule.

In which case, this should try to make this less about circuit and more about the domain of this pallet, maybe PendingLedgerItem. Which would be functionality that allows a transfer to be promoted to the queue and thus handled by the clock.